score-xml/src/main/java/io/cloudslang/content/xml/services/ValidateService.java

```diff
@@ -2,57 +2,81 @@
 
 import io.cloudslang.content.xml.entities.inputs.CommonInputs;
 import io.cloudslang.content.xml.entities.inputs.CustomInputs;
-import io.cloudslang.content.xml.utils.Constants;
+import io.cloudslang.content.xml.entities.Constants;
 import io.cloudslang.content.xml.utils.XmlUtils;
 import org.apache.commons.lang3.StringUtils;
+import org.w3c.dom.Document;
 import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
 
 import javax.xml.XMLConstants;
+import javax.xml.transform.TransformerException;
 import javax.xml.transform.stream.StreamSource;
 import javax.xml.validation.Schema;
 import javax.xml.validation.SchemaFactory;
 import javax.xml.validation.Validator;
-import java.io.StringReader;
+import java.io.*;
 import java.util.HashMap;
 import java.util.Map;
 
 /**
  * Created by markowis on 03/03/2016.
  */
 public class ValidateService {
-    public Map<String, String> execute(CommonInputs commonInputs, CustomInputs customInputs){
+    public Map<String, String> execute(CommonInputs commonInputs, CustomInputs customInputs) {
         Map<String, String> result = new HashMap<>();
 
         try {
-            String xmlDocument = commonInputs.getXmlDocument();
-            String xsdDocument = customInputs.getXsdDocument();
+            String xmlDocument;
+            String xsdDocument;
+            if (Constants.XML_PATH.equalsIgnoreCase(commonInputs.getXmlDocumentSource())) {
+                Document doc = XmlUtils.createDocumentFromFile(commonInputs.getXmlDocument(), commonInputs.getSecureProcessing());
+                xmlDocument = getStringValueOfDocument(doc);
+            } else if (Constants.XML_URL.equalsIgnoreCase(commonInputs.getXmlDocumentSource())) {
+                xmlDocument = XmlUtils.createXmlDocumentFromUrl(commonInputs);
+                xmlDocument = xmlDocument.replaceAll("<!DOCTYPE[^>]*>\n", "");
+            } else {
+                xmlDocument = commonInputs.getXmlDocument();
+            }
 
+            if (Constants.XSD_PATH.equalsIgnoreCase(customInputs.getXsdDocumentSource())) {
+                Document doc = XmlUtils.createDocumentFromFile(customInputs.getXsdDocument(), commonInputs.getSecureProcessing());
+                xsdDocument = getStringValueOfDocument(doc);
+            } else {
+                xsdDocument = customInputs.getXsdDocument();
+            }
             XmlUtils.parseXML(xmlDocument, commonInputs.getSecureProcessing());
-            result.put(Constants.OutputNames.RETURN_RESULT, Constants.SuccessMessages.PARSING_SUCCESS);
+            result.put(Constants.Outputs.RETURN_RESULT, Constants.SuccessMessages.PARSING_SUCCESS);
 
-            if(xsdDocument != null && StringUtils.isNotBlank(xsdDocument)) {
-                validateAgainstXsd(xmlDocument, customInputs.getXsdDocument());
-                result.put(Constants.OutputNames.RETURN_RESULT, Constants.SuccessMessages.VALIDATION_SUCCESS);
+            if (xsdDocument != null && StringUtils.isNotBlank(xsdDocument)) {
```

Comment:
StringUtils.isNotBlank already contains a null check, so the first part of this if is redundant.